core/src/OpenMPTarget/Kokkos_OpenMPTarget_Parallel_Common.hpp

```diff
@@ -33,9 +33,12 @@ struct ParallelReduceCopy {
   static void memcpy_result(PointerType dest, PointerType src, size_t size,
                             bool ptr_on_device) {
     if (ptr_on_device) {
```

Comment:
Just to mention, everywhere else you have 
```
if(0<size)
  if(ptr_on_device)
```
While here you have flipped the order of the `if` conditions. Should not make a difference but just wanted to mention.